src/main/java/com/ethanyidong/bunny/ParsedCommand.java

```diff
@@ -0,0 +1,73 @@
+package com.ethanyidong.bunny;
+
+import com.ethanyidong.bunny.arg.CommandValidator;
+import com.ethanyidong.bunny.arg.InvalidArgumentException;
+import com.ethanyidong.bunny.arg.InvalidCommandException;
+
+import java.util.HashMap;
+
+public class ParsedCommand {
+    private CommandType command;
+    private String positionalArgument;
+    private HashMap<String, String> flagArguments;
+    public ParsedCommand(String command) {
+        String[] commandAndArguments = command.split(" ", 2);
+        this.command = extractCommandType(commandAndArguments[0]);
+
+        String arguments;
+        if(commandAndArguments.length == 2) {
+            arguments = commandAndArguments[1];
+        } else {
+            return;
+        }
+
+        String[] positionalAndFlagArguments = arguments.split("/");
+        this.positionalArgument = extractPositionalArgument(positionalAndFlagArguments);
+        this.flagArguments = extractFlagArguments(positionalAndFlagArguments);
+
+    }
+
+    private static CommandType extractCommandType(String command) {
+        return CommandType.fromString(command);
+    }
+
+    private static String extractPositionalArgument(String[] positionalAndFlagArgument) {
+        return positionalAndFlagArgument[0].trim();
+    }
+
+    private static HashMap<String, String> extractFlagArguments(String[] positionalAndFlagArguments) {
+        HashMap<String, String> ret = new HashMap<>();
```

Comment:
Maybe this variable should have a clearer name. `ret` is not an English word so it's not immediately clear to me what it means.